core/domain/state_domain.py

```diff
@@ -149,6 +149,43 @@ def validate(self, interaction, exp_param_specs_dict):
 
         self.outcome.validate()
 
+    def get_all_html_content_strings(self):
+        """Get all html content strings in the AnswerGroup.
+
+        Returns:
+            list(str). The list of all html content strings in the interaction.
+        """
+        html_list = []
+
+        outcome_html = self.outcome.feedback.html
+        html_list = html_list + [outcome_html]
+        for rule_spec in self.rule_specs:
```

Comment:
Do the following instead, for future-proofing:

- Make a constant with several lists of tuples mapping type to (interaction id, rule type, param name). E.g. {'ListOfSetsOfHtmlString': [('DragAndDropSortInput', 'IsEqualToOrdering', 'x'), ...], ...}
- Have a backend test to rederive that constant from extensions/rule_templates.json and verify it's correct.
- In this method, iterate over that constant to figure out the assembly of the html list.

Reply:
@seanlip , we had discussed earlier that this is currently not possible because this method will be used for draft changes too and draft changes don't have access to interaction id.
pls refer [comment](https://github.com/oppia/oppia/pull/9390#discussion_r433004892)

Reply:
perhaps I can add a docstring explaining the reason.

Reply:
You can still use the approach mentioned above. Just look at the rule types and param names (as you're doing here), and add a comment explaining why we aren't using the "interaction id" part. The main thing is to make it more declarative rather than have lots of special-casing for particular types of interactions within the main code.

Reply:
done